pytroll_collectors/segments.py

```diff
@@ -406,7 +408,18 @@ def is_relevant(self, message):
         if mask in slot_pattern['received_files']:
             logger.debug("File already received")
             should_be_added = False
-        if mask not in slot_pattern['all_files']:
+        if mask in slot_pattern['all_files']:
+            pass
+        elif message.slot_time is not None:
+            # File probably within tolerance of existing slot
+            metadata_new = metadata.copy()
+            metadata_new[message._time_name] = message.slot_time  # Use slot time insted of file_time
+            mask_temp = message.pattern.parser.globify(metadata_new)
+            if mask_temp not in slot_pattern['all_files']:
+                logger.debug("%s not in %s", mask, slot_pattern['all_files'])
+                logger.debug("%s not in %s", mask_temp, slot_pattern['all_files'])
+                should_be_added = False
+        else:
```

Comment:
And then this could be replaced by a function call like:
`should_be_added = check_if_file_is_within_tolerance_of_existing_slot(message, metadata, mask)`
And you could then skip the comments! Adding comments in code is usually a sign that the code doesn't read for itself.

Reply:
I agree with @adybbroe here, this could be refactored a bit.